rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,586 @@
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ListNamespacesResponse'
+        '401':
+          description: Unauthorized
+    post:
+      tags:
+        - Catalog API
+      summary: Create a namespace
+      description: Create a namespace, with an optional set of properties. The server might also add properties, such as last_modified_time etc.
+      operationId: createNamespace
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateNamespaceRequest'
+        required: true
+      responses:
+        '200':
+          description: OK
+        '401':
+          description: Unauthorized
+        '409':
+          description: Conflict (AlreadyExistsException)
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NamespaceAlreadyExistsError'
+              example:
+                '{ error: { message: "Namespace already exists", type: "AlreadyExistsException", code: 40901 }'
+
+  /namespaces/{namespace}:
+    parameters:
+      - name: namespace
+        in: path
+        required: true
+        schema:
+          type: string
+    get:
+      tags:
+        - Catalog API
+      summary: Load the metadata properties for a namespace
+      operationId: loadNamespaceMetadata
+      description: Return all stored metadata properties for a given namespace
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/GetNamespaceResponse'
+        '404':
+          description: Not Found (NoSuchNamespaceException)
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NoSuchNamespaceError'
+              example:
+                '{ error: { message: "Namespace does not exist", type: "NoSuchNamespaceException", code: 40401 }'
+    head:
+      tags:
+        - Catalog API
+      summary: Check if a namespace exists
+      operationId: namespaceExists
+      description: Check if a namespace exists.
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/IcebergResponseObject'
+              example: '{ "data": { "exists": false } }'
+        '401':
+          description: Unauthorized
+    delete:
+      tags:
+        - Catalog API
+      summary: Drop a namespace from the catalog. Namespace must be empty.
+      operationId: dropNamespace
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/IcebergResponseObject'
+              example: '{ "data": { "dropped": true } }'
```

Comment:
will this return `dropped = false` when calling this endpoint again after the resource has been deleted?
I think this endpoint is also missing a 404 status code if the resource doesn't exist, unless we want to return dropped=true/false depending on whether the resource was deleted or not.

`DELETE` is supposed to be idempotent, but I think it's ok to actually return dropped=true/false as long as the resource was deleted on the first call and additional calls of this endpoint won't change the state on the server, but return dropped=false

Reply:
Hmmm. I guess it would be hard to remain idempotent about the call to drop (vs just does not exist) and then returned `dropped: false`.

In that case, considering somebody might have multiple servers involved etc, do you have alternatives that you suggest?

Reply:
I think this specific use case when deleting something isn't super well defined in REST. One could argue that the cleanest approach would be to perform the DELETE when the resource exists and error out with a 404 if it doesn't exist.
On the other hand you could still model it the same way as you did. The outcome on the server is the same, the resource is deleted (hence the endpoint is idempotent). The only difference is how you model the case when the resource doesn't exist (anymore)